src/main/java/hudson/tasks/junit/TestResultSummary.java

```diff
@@ -0,0 +1,41 @@
+package hudson.tasks.junit;
+
+import java.io.Serializable;
+
+/**
+ * CPS-friendly summary of test results that
+ * serializes counts
+ */
+public class TestResultSummary implements Serializable {
+
+
+    private int failCount;
+    private int passCount;
+    private int skipCount;
+    private int totalCount;
+
+    public TestResultSummary(TestResult result) {
+        result.tally();
+        failCount = result.getFailCount();
+        passCount = result.getPassCount();
+        skipCount = result.getSkipCount();
+        totalCount = result.getTotalCount();
+    }
+
+    public int getFailCount() {
```

Comment:
All these getters would need to be `@Whitelisted` to be accessible by default from a sandboxed script.